kolibri/core/assets/src/api_resource.js

```diff
@@ -105,46 +251,71 @@ class Collection {
-          // paginated data! Check to see if it is paginated.
-          if (typeof response.entity.results !== 'undefined') {
-            // Paginated objects have 'results' as their results object so interpret this as
-            // such.
-            this.set(response.entity.results);
-            this.pageCount = Math.ceil(response.entity.count / this.pageSize);
-          } else {
-            // It's all gone a bit Pete Tong.
-            logging.debug('Data appears to be malformed', response.entity);
-          }
+          this.synced = false;
+          client({ path: this.url, params }).then((response) => {
+            // Reset current models to only include ones from this fetch.
+            this.models = [];
+            this._model_map = {};
+            // Set response object - an Array - on the Collection to record the data.
+            // First check that the response *is* an Array
+            if (Array.isArray(response.entity)) {
+              this.set(response.entity);
+            } else {
+              // If it's not, there are two possibilities - something is awry, or we have received
+              // paginated data! Check to see if it is paginated.
+              if (typeof response.entity.results !== 'undefined') {
+                // Paginated objects have 'results' as their results object so interpret this as
+                // such.
+                this.set(response.entity.results);
+                this.pageCount = Math.ceil(response.entity.count / this.pageSize);
+              } else {
+                // It's all gone a bit Pete Tong.
+                logging.debug('Data appears to be malformed', response.entity);
+              }
+            }
+            // Mark that the fetch has completed.
+            this.synced = true;
+            this.models.forEach((model) => {
+              model.synced = true; // eslint-disable-line no-param-reassign
+            });
+            // Return the data from the models, not the models themselves.
+            resolve(this.data);
+            // Clean up the reference to this promise
+            this.promises.splice(this.promises.indexOf(promise), 1);
+          }, (response) => {
+            logging.error('An error occurred', response);
+            reject(response);
+            // Clean up the reference to this promise
+            this.promises.splice(this.promises.indexOf(promise), 1);
+          });
         }
-        // Mark that the fetch has completed.
-        this.synced = true;
-        this.models.forEach((model) => {
-          model.synced = true; // eslint-disable-line no-param-reassign
-        });
-        // Return the data from the models, not the models themselves.
-        resolve(this.data);
+      },
+      (reason) => {
+        reject(reason);
+      });
+    });
+    this.promises.push(promise);
+    return promise;
+  }
+
+  getCurrentFacility() {
```

Comment:
This should not be added to the `Collection` class as a whole - seems rather specific.

For MVP at least, I feel like this would be something that is better to be bootstrapped into the Django View and set as a global variable.